### ISSUE TITLE
why doesnt NativeFileChooser::show return the underlying result of the fltk show call

### ISSUE THREAD

**Issue by pm100** [fltk-rs/fltk-rs#1513]
The c++ docs have this sample

```
switch ( fnfc.show){
  case -1: printf("ERROR: %s\n", fnfc.errmsg);    break;  // ERROR
  case  1: printf("CANCEL\n");                      break;  // CANCEL
  default: printf("PICKED: %s\n", fnfc.filename; break;  // FILE CHOSEN
}
```

This is not doable from rust code.

I think I have to look for a blank file name, meaning 'cancel', I cant look for `Some(_)` from `error_message()` becasue that returns Some("No Error") if there is no error (!)

And I cant even do a simple 'filename().len()>0' I have to do

```
       filename.to_string_lossy().to_string().is_empty()
```

according to the editor sample. 

At least filename could return `Option<Path>`

**Comment by MoAlyousef** [comment]
At the time of writing the bindings for Fl_Native_File_Chooser, I didn't know of a better way to model the return values of show(), which are -1 for failed, 0 for success, 1 for cancelled. 
I still don't think it would be best modeled using a Result type. -1 can be an Err(errmsg()), what about 1?
Is a cancellation an error?

Regarding ` filename.to_string_lossy().to_string().is_empty()`, you can use ` filename.as_os_str().is_empty()` or `filename.exists()`.
`Option<Path>` doesn't work as a return type since Path is unsized. It could be `Option<PathBuf>` though.

P.S. Maybe returning an Enum value indicating the result of show() is more appropriate.

**Comment by MoAlyousef** [comment]
I have added a try_show() method and have updated the editor example to show it.
Once FLTK 1.4 is officially released, I'll start working on version 2.0 where show() would return a Result instead.
